cli/commands/network.py

```diff
@@ -0,0 +1,160 @@
+import click
+import logging
+import multiprocessing as mp
+import os
+import time
+import sys
+
+from pathlib import Path
+
+from SimulaQron.settings import Settings
+from SimulaQron.virtNode.virtual import backEnd
+from SimulaQron.run.startCQC import main as start_cqc
+from SimulaQron.run.startCQCLog import main as start_cqc_log
+from SimulaQron.run.startNode import main as start_node
+from SimulaQron.configFiles import construct_node_configs, construct_topology_config
+
+
+def _load_node_names(path):
+    """Load list of nodes from Nodes.cfg file
+
+    :path: pathlib.Path pointing to Nodes.cfg file"""
+    with path.open() as f:
+        return [line.strip() for line in f.readlines()]
+
+
+_common_options = [
+    click.option(
+        "--netsim",
+        envvar="NETSIM",
+        help="Network simulation root (default: $NETSIM)",
+        type=click.Path(exists=True, file_okay=False, dir_okay=True),
+    ),
+    click.option(
+        "-N",
+        "--nrnodes",
+        help="Number of nodes to start \n(WARNING: overwites existing config files)",
+        type=click.INT,
+        default=None,
+    ),
+    click.option(
+        "-n",
+        "--nodes",
+        help="Comma separated list of nodes to start \n(WARNING: overwites existing config files)",
+        type=click.STRING,
+        default=None,
+    ),
+    click.option(
+        "-t",
+        "--topology",
+        help="Topology of network \n(WARNING: overwites existing config files)",
+        type=click.STRING,
+        default=None,
+    ),
+]
+
+
+def common_options(func):
+    """Decorator to add all common options to a click command"""
+    for option in reversed(_common_options):
+        func = option(func)
+    return func
+
+
+@click.group()
+def network():
+    """Commands for managing SimulaQron networks."""
+    pass
+
+
+@network.command()
+@common_options
+def start_all(netsim, nrnodes, nodes, topology):
```

Comment:
Would it be possible to also add a "kill" command? For example if I start the network as a background process `./cli/main network start_all &` then it would be convenient to be able to kill it without having to find the process number manually.